restapi/api/reserve_endpoints_test.go

```diff
@@ -122,9 +124,147 @@ func TestValidateDeviceNotReserved(t *testing.T) {
 	validateNotReserved(t, responseRecorder)
 }
 
+func TestValidateMiddlewareHeaderMissing(t *testing.T) {
+	r = setupRouter()
+	responseRecorder := httptest.NewRecorder()
+
+	launchAppRequest, err := http.NewRequest("POST", "/"+randomDeviceUDID+"/launch?bundleID=com.apple.Preferences", nil)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	r.ServeHTTP(responseRecorder, launchAppRequest)
+	require.Equal(t, http.StatusBadRequest, responseRecorder.Code, "Code should be BadRequest if X-GO-IOS-RESERVE header is missing")
+
+	var response GenericResponse
+	responseData, _ := ioutil.ReadAll(responseRecorder.Body)
```

Comment:
ioutil.ReadAll is deprecated. Just use io.ReadAll please.

Reply:
done